libs/ballot-interpreter/src/hmpb-rust/diagnostic.rs

```diff
@@ -0,0 +1,182 @@
+use std::path::PathBuf;
+
+use image::{GenericImageView, GrayImage};
+use imageproc::contrast::otsu_level;
+use logging_timer::time;
+use rayon::prelude::{IntoParallelRefIterator, ParallelIterator};
+use types_rs::geometry::Rect;
+
+use crate::{
+    ballot_card::load_ballot_scan_bubble_image,
+    debug::{draw_diagnostic_cells, ImageDebugWriter},
+    image_utils::{ratio, BLACK},
+};
+
+const FAIL_SCORE: f32 = 0.05;
```

Comment:
Note that you can bring this down to 0.03 and all tests will pass, but I wanted to give it some leeway due to unpredictability of field data.